modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -5067,7 +5067,7 @@ bool GDScriptAnalyzer::is_type_compatible(const GDScriptParser::DataType &p_targ
 		if (valid && p_target.builtin_type == Variant::ARRAY && p_source.builtin_type == Variant::ARRAY) {
 			// Check the element type.
 			if (p_target.has_container_element_type() && p_source.has_container_element_type()) {
-				valid = p_target.get_container_element_type() == p_source.get_container_element_type();
+				valid = is_type_compatible(p_target.get_container_element_type(), p_source.get_container_element_type(), false, p_source_node);
```

Comment:
This is not an error, element types must be equal, supertypes are not supported. I have a [draft](https://github.com/dalexeev/godot/blob/1b596113a112e99fc3769cc9afa6485c81ed1e6c/modules/gdscript/gdscript_analyzer.cpp#L5084-L5092) with the `is_type_compatible()` refactoring in which I added a comment about it.